### ISSUE TITLE
semicolon rule not working as expected

### ISSUE THREAD

**Issue by xahon** [palantir/tslint#4794]
### Bug Report

- __TSLint version__:     "tslint": "^5.18.0",
- __TypeScript version__:     "typescript": "3.5.3"
- __Running TSLint via__: Rider/WebStorm

#### TypeScript code being linted

```ts
class A {
    private a: number;
    private b: number // no semicolon and error

    private method = () => {
        this.a = 20;
    }; // unncessary?

    private method2 = () => {
        this.a = 20;
    } // no semicolon and no error o_O

}
```

with `tslint.json` configuration:

```json
// You can modify the configuration here with your own.
// Make sure that your json is valid, and if you run into
// other issues, try checking the console for errors.

{
    "extends": ["tslint:recommended"],
    "rules": {
        "no-console": false,
        "semicolon": true
    } // add additional rules and their configuration
}
```

#### Actual behavior
weird

#### Expected behavior
semicolons everywhere not causing errors

**Comment by JoshuaKGoldberg** [comment]
Hi @xahon! This is the intended behavior of the `semicolon` rule by default. Per https://palantir.github.io/tslint/rules/semicolon/, you can configure it to require semicolons to `"always"` or `"never"` exist. Additionally:

> The following arguments may be optionally provided:
>
> * "ignore-interfaces" skips checking semicolons at the end of interface members.
> * "ignore-bound-class-methods" skips checking semicolons at the end of bound class methods.
> * "strict-bound-class-methods" disables any special handling of bound class methods and treats them as any other assignment. This option overrides "ignore-bound-class-methods".

**Comment by xahon** [comment]
@JoshuaKGoldberg there is no option to ignore field definitions
```ts
class A {
    private a: number;
    private b: number // no semicolon and error
    ...
```
